core/src/retransmit_stage.rs

```diff
@@ -243,34 +242,17 @@ fn check_if_already_received(
 }
 
 // Returns true if turbine retransmit peers patch (#14565) is enabled.
-fn enable_turbine_retransmit_peers_patch(
-    shred_slot: Slot,
-    bank_forks: &RwLock<BankForks>,
-    cache: &mut LruCache<Slot, Arc<Bank>>,
-) -> bool {
-    let bank: Arc<Bank> = match cache.get(&shred_slot) {
-        Some(bank) => bank.clone(),
-        None => {
-            let bank_forks = bank_forks.read().unwrap();
-            match bank_forks.get(shred_slot) {
-                Some(bank) => {
-                    cache.put(shred_slot, bank.clone());
-                    bank.clone()
-                }
-                None => bank_forks.root_bank(),
-            }
-        }
-    };
-    let feature_slot = bank
+fn enable_turbine_retransmit_peers_patch(shred_slot: Slot, root_bank: &Bank) -> bool {
+    let feature_slot = root_bank
         .feature_set
         .activated_slot(&feature_set::turbine_retransmit_peers_patch::id());
     match feature_slot {
         None => false,
         Some(feature_slot) => {
-            let epoch_schedule = bank.epoch_schedule();
+            let epoch_schedule = root_bank.epoch_schedule();
             let feature_epoch = epoch_schedule.get_epoch(feature_slot);
             let shred_epoch = epoch_schedule.get_epoch(shred_slot);
-            feature_epoch < shred_epoch
+            feature_epoch + 1 < shred_epoch
```

Comment:
Sorry, I think I read the discord comment wrong, I think  this should be a `<=`, or change `feature_epoch + 1` to `feature_epoch + 2`.

So if the feature activates on epoch `N`, then  we want to start returning `true` here on epoch  `N + 1`